src/main/java/org/ice4j/socket/MultiplexingXXXSocketSupport.java

```diff
@@ -188,6 +188,24 @@ else if (logger.isLoggable(Level.WARNING))
         }
     }
 
+    /**
+     * The max number of {@link DatagramPacket}'s stored in the receive pool.
+     */
+    private static final int RECEIVE_DATAGRAM_PACKETS_POOL_SIZE = 64;
+
+    /**
+     * The buffer size of pooled datagram packet used to read from
+     * underlying socket.
+     */
+    private static final int POOLED_DATAGRAM_PACKET_SIZE = 1500;
```

Comment:
Do you know how is this value related to the 'receiveBufferSize' ? do you think it would be a problem if the buffer size would be bigger ? I see you've chosen Ethernet's MTU as default which I guess should be fine most of the time, but what bothers me is that this socket allows to adjust the buffer's size.

Reply:
> Do you know how is this value related to the 'receiveBufferSize' ?

This constant limits the max datagram packet size read from Java’s DatagramSocket.
`receiveBufferSize` controls the size of kernel buffer for socket (SO_RECVBUF), but in this case the socket is kind of virtual and this value controls the max total number of bytes stored across all datagrams stored in receive buffer - which is in this case just List<DatagramPakcet>(until #167 merged). As I understand it - it just a programmatic simulation of kernel behavior which limits the number of bytes socket can have “unread”, before start dropping the most old packets.


> do you think it would be a problem if the buffer size would be bigger ?

If I understand your question correctly, the no, since the `receiveBufferSize` and `POOLED_DATAGRAM_PACKET_SIZE` controls different things.


It probably make sense to have such many packets in pool as `receiveBufferSize` allow in receive buffer, but until #167 merged the socket receive buffer  is just a `List<DatagramPacket>` which makes it inconvinient to add pooling directly inside socket receive buffer.